server/aesdsocket.c

```diff
@@ -163,17 +423,405 @@ static int daemon_init()
+    {
+        printf("\nError: Failed read(). Error code: %d\n", errno);
+        // Syslog the error into the syslog file in /var/log
+        syslog(LOG_ERR, "Error: Failed read(). Error code: %d", errno);
+        // return -1;
+        goto err_handle;
+    }
+
+#if DEBUG
+    printf("Data Len Transmitted: %d\n", socket_file_fd_len);
+    for(i=0;i<socket_file_fd_len;i++)
+    {
+        printf("%c", *(tx_storage_buffer+i));
+    }
+    printf("\n"); // It seems line the system was buffering the printf so absense of new line was making it buffer and printing only the next time new line was met
+    // Refer to https://stackoverflow.com/questions/39180642/why-does-printf-not-produce-any-output
+#endif
+
+    /*********************************************************************************************************
+                                    Sending Data from server to client
+    **********************************************************************************************************/
+    // The system calls send(), sendto(), and sendmsg() are used to
+    // transmit a message to another socket.
+    // ssize_t send(int sockfd, const void *buf, size_t len, int flags);
+    // manpage: https://man7.org/linux/man-pages/man2/send.2.html
+    int tx_data_len;
+    tx_data_len = send(client_socket_fd, tx_storage_buffer, socket_file_fd_len, 0);
+    if(tx_data_len == -1) // returns -1 on error else number of bytes sent
+    {
+        printf("\nError: Failed send(). Error code: %d\n", errno);
+        // Syslog the error into the syslog file in /var/log
+        syslog(LOG_ERR, "Error: Failed send(). Error code: %d", errno);
+        // return -1;
+        goto err_handle;
+    }
+
+    // The mutex object referenced by mutex shall be locked by a call to
+    // pthread_mutex_lock() that returns zero or [EOWNERDEAD].  If the
+    // mutex is already locked by another thread, the calling thread
+    // shall block until the mutex becomes available.
+    // int pthread_mutex_lock(pthread_mutex_t *mutex);
+    //  manpage: https://man7.org/linux/man-pages/man3/pthread_mutex_lock.3p.html
+    status = pthread_mutex_unlock(&mutex);
+    if(status != 0) // returns non zero on error
+    {
+        printf("\nError: Failed pthread_mutex_unlock(). Error code: %d\n", errno);
+        // Syslog the error into the syslog file in /var/log
+        syslog(LOG_ERR, "Error: Failed pthread_mutex_unlock(). Error code: %d", errno);
+        // return -1;
+    }
+
+err_handle:
+    free(tx_storage_buffer);
+    free(rx_storage_buffer);
+
+    printf("Connection Closed with %s :: %d\n", ip, p);
+    // Syslog the info into the syslog file in /var/log
+    syslog(LOG_INFO, "Connection Closed with %s :: %d\n", ip, p);
+
+    thread_arg->thread_complete = true;
+
+    return NULL;
+}
+
+
+void ll_push(node_t **head_ref, node_t *node)
+{
+    node->next_node = *head_ref;
+    *head_ref = node;
+}
+
+void clean_ll(node_t** head, int is_flag)
```

Comment:
@nihalthirunakarasu I like the plan to use this to indicate joining without thread complete.  A better name and boolean type would be good, along with some comments.